src/Access/ContextAccess.cpp

```
@@ -513,6 +513,8 @@ std::shared_ptr<const EnabledQuota> ContextAccess::getQuota() const
 
 std::optional<QuotaUsage> ContextAccess::getQuotaUsage() const
 {
+    if (getQuota() == nullptr) /// Detected by fuzzer
```

1. Can we write the code like this

```cpp
auto quota = getQuota();
if (!quota)
    return {};
else
    quota->getUsage();
```

2. Returning `std::nullopt` may cause follow-up problems further up the stack. Perhaps we should throw an exception similar to l. 491? If _only_ the fuzzer runs into this path it can be changed to handle this situation gracefully.

---

I returned `nullopt` because the routine had the capability of `optional` return. I can throw an exception. The fuzzer is well designed to handle all exceptions, I will anyways again test with throwing an exception.

---

Turns out that throwing a exception of type `LOGICAL_ERROR` will still cause the fuzzer to abort (because of https://github.com/ClickHouse/ClickHouse/blob/a7f922cea2ce628137d2ca0e823e56fe090d04aa/src/Common/Exception.cpp#L68. Only `LOGICAL_ERROR` makes sense I think. 

I have retained the return of `nullopt` since that was in place.